atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbKeyValueServiceConfig.java

```diff
@@ -29,9 +42,53 @@ public DbSharedConfig shared() {
         return ImmutableDbSharedConfig.builder().build();
     }
 
+    public abstract ConnectionConfig connection();
+
     @Value.Derived
     public abstract Supplier<DbTableFactory> tableFactorySupplier();
 
-    public abstract SqlConnectionSupplier sqlConnectionSupplier();
+    @Value.Default
+    public Supplier<SqlConnectionSupplier> sqlConnectionSupplier() {
+        return () -> createConnectionSupplier();
```

Comment:
If we have this in the config is has to be a supplier (or else people will start spawning connection pools whenever they request access to the connection pool).  That said, this code will completely go away (in a revised version of this PR) because as we discussed early I'm moving this in into the constructor.